File: backend/app/api/v1/learning_path.py
from fastapi import APIRouter, Path

from app.core.response import error_response, success_response
from app.schemas.learning_path import LearningPathGenerateRequest, LearningTaskStatusUpdateRequest
from app.services.learning_path_service import LearningPathService

router = APIRouter()
learning_path_service = LearningPathService()


@router.post("/learning-paths/generate")
async def generate_learning_path(payload: LearningPathGenerateRequest):
    try:
        plan = await learning_path_service.generate_learning_path(payload)
    except Exception as exc:
        return error_response(f"学习路径生成失败：{exc}")
    return success_response(plan.learning_path)


@router.get("/users/{userId}/learning-paths")
def list_user_paths(user_id: str = Path(alias="userId")):
    return success_response(learning_path_service.list_user_paths(user_id))


@router.get("/learning-paths/{pathId}")
def get_learning_path(path_id: str = Path(alias="pathId")):
    path = learning_path_service.get_learning_path(path_id)
    if path is None:
        return error_response("学习路径不存在", code=404)
    return success_response(path)


@router.put("/learning-paths/{pathId}")
def update_learning_path(payload: dict, path_id: str = Path(alias="pathId")):
    path = learning_path_service.update_learning_path(path_id, payload)
    if path is None:
        return error_response("学习路径不存在", code=404)
    return success_response(path)


@router.get("/learning-paths/{pathId}/tasks")
def list_path_tasks(path_id: str = Path(alias="pathId")):
    tasks = learning_path_service.list_path_tasks(path_id)
    return success_response(tasks)


@router.put("/learning-tasks/{taskId}/status")
def update_task_status(payload: LearningTaskStatusUpdateRequest, task_id: str = Path(alias="taskId")):
    task = learning_path_service.update_task_status(task_id, payload)
    if task is None:
        return error_response("学习任务不存在", code=404)
    return success_response(task)

File: frontend/src/api/modules/learningPath.ts
import { request } from "@/api/client";
import type {
  LearningPath,
  LearningPathGenerateRequest,
  LearningTask,
  LearningTaskStatusUpdateRequest
} from "@/types/learningPath";

const learningPathGenerateTimeout = 3 * 60 * 1000;

export const learningPathApi = {
  generateLearningPath(payload: LearningPathGenerateRequest) {
    return request<LearningPath>({
      method: "POST",
      url: "/learning-paths/generate",
      data: payload,
      timeout: learningPathGenerateTimeout
    });
  },
  generatePath(payload: LearningPathGenerateRequest) {
    return request<LearningPath>({
      method: "POST",
      url: "/learning-paths/generate",
      data: payload,
      timeout: learningPathGenerateTimeout
    });
  },
  getUserLearningPaths(userId: string) {
    return request<LearningPath[]>({ method: "GET", url: `/users/${userId}/learning-paths` });
  },
  listUserPaths(userId: string) {
    return request<LearningPath[]>({ method: "GET", url: `/users/${userId}/learning-paths` });
  },
  getLearningPath(pathId: string) {
    return request<LearningPath>({ method: "GET", url: `/learning-paths/${pathId}` });
  },
  getPath(pathId: string) {
    return request<LearningPath>({ method: "GET", url: `/learning-paths/${pathId}` });
  },
  updatePath(pathId: string, payload: Partial<LearningPath>) {
    return request<LearningPath>({ method: "PUT", url: `/learning-paths/${pathId}`, data: payload });
  },
  updateLearningPath(pathId: string, payload: Partial<LearningPath>) {
    return request<LearningPath>({ method: "PUT", url: `/learning-paths/${pathId}`, data: payload });
  },
  getLearningTasks(pathId: string) {
    return request<LearningTask[]>({ method: "GET", url: `/learning-paths/${pathId}/tasks` });
  },
  listPathTasks(pathId: string) {
    return request<LearningTask[]>({ method: "GET", url: `/learning-paths/${pathId}/tasks` });
  },
  updateLearningTaskStatus(taskId: string, payload: LearningTaskStatusUpdateRequest) {
    return request<LearningTask>({ method: "PUT", url: `/learning-tasks/${taskId}/status`, data: payload });
  },
  updateTaskStatus(taskId: string, payload: LearningTaskStatusUpdateRequest) {
    return request<LearningTask>({ method: "PUT", url: `/learning-tasks/${taskId}/status`, data: payload });
  }
};
